app.js

```diff
@@ -1,18 +1,51 @@
 const { App } = require("@slack/bolt");
 const express = require("express");
-
 const webserver = express();
 
+const app = new App({
+  token: process.env.BOT_USER_OAUTH_ACCESS_TOKEN,
+  socketMode: true,
+  appToken: process.env.APP_TOKEN,
+});
+
 webserver.get("/", (req, res) => {
   res.send("Gratibot is running!");
 });
 
-webserver.listen(process.env.PORT || 3000);
+webserver.get("/health", async (req, res) => {
+  const status_checks = {};
 
-const app = new App({
-  token: process.env.BOT_USER_OAUTH_ACCESS_TOKEN,
-  socketMode: true,
-  appToken: process.env.APP_TOKEN,
+  // Check Slack API
+  try {
+    const slack_api_status = await app.client.api.test();
+    if (slack_api_status.ok) {
+      status_checks.slack_api = "OK";
+    }
+  } catch (e) {
+    status_checks.slack_api = e.message;
+  }
+
+  // Check Slack Auth
+  try {
+    const slack_auth_status = await app.client.auth.test();
```

Comment:
how frequently do these health checks run?  it might be a bit much if these were done every second, for example

Reply:
Once a minute.

And note, that both of these calls are tier 4 (One is labeled special, which I admittedly don't understand the reason for), and would theoretically support up to 100 requests a minute.
https://api.slack.com/methods/auth.test
https://api.slack.com/methods/api.test

Not that I would want this bot to run that much.